test/Microsoft.AspNetCore.Sockets.Tests/HttpConnectionDispatcherTests.cs

```diff
@@ -40,41 +39,61 @@ public class HttpConnectionDispatcherTests
             Assert.Equal(id, state.Connection.ConnectionId);
         }
 
-        [Fact]
-        public async Task SendingToUnknownConnectionIdThrows()
+        [Theory]
+        [InlineData("/send")]
+        [InlineData("/sse")]
+        [InlineData("/poll")]
+        [InlineData("/ws")]
+        public async Task EndpointsThatAcceptConnectionId404WhenUnknownConnectionIdProvided(string path)
         {
             var manager = new ConnectionManager();
             var dispatcher = new HttpConnectionDispatcher(manager, new LoggerFactory());
-            var context = new DefaultHttpContext();
-            var services = new ServiceCollection();
-            services.AddSingleton<TestEndPoint>();
-            context.RequestServices = services.BuildServiceProvider();
-            context.Request.Path = "/send";
-            var values = new Dictionary<string, StringValues>();
-            values["id"] = "unknown";
-            var qs = new QueryCollection(values);
-            context.Request.Query = qs;
-            await Assert.ThrowsAsync<InvalidOperationException>(async () =>
+
+            using (var strm = new MemoryStream())
             {
+                var context = new DefaultHttpContext();
+                context.Response.Body = strm;
+
+                var services = new ServiceCollection();
+                services.AddSingleton<TestEndPoint>();
+                context.RequestServices = services.BuildServiceProvider();
+                context.Request.Path = path;
+                var values = new Dictionary<string, StringValues>();
+                values["id"] = "unknown";
+                var qs = new QueryCollection(values);
+                context.Request.Query = qs;
+
                 await dispatcher.ExecuteAsync<TestEndPoint>("", context);
-            });
+
+                Assert.Equal(404, context.Response.StatusCode);
+                await strm.FlushAsync();
+                Assert.Equal("No Connection with that ID", Encoding.UTF8.GetString(strm.ToArray()));
+            }
         }
 
-        [Fact]
-        public async Task SendingWithoutConnectionIdThrows()
+        [Theory]
+        [InlineData("/send")]
+        [InlineData("/sse")]
+        [InlineData("/poll")]
+        public async Task EndpointsThatRequireConnectionId400WhenNoConnectionIdProvided(string path)
```

Comment:
The problem is, that requires successfully establishing a connection in this unit test, which we don't really have working because of how we mock things out (using DefaultHttpContext)